### ISSUE TITLE
ProtocolServer/LibWeb: Support for duplicate field names in headers

### ISSUE THREAD

**Issue by squeek502** [SerenityOS/serenity#2091]
From https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2:

> Multiple message-header fields with the same field-name MAY be present in a message if and only if the entire field-value for that header field is defined as a comma-separated list [i.e., #(values)]. It MUST be possible to combine the multiple header fields into one "field-name: field-value" pair, without changing the semantics of the message, by appending each subsequent field-value to the first, each separated by a comma. The order in which header fields with the same field-name are received is therefore significant to the interpretation of the combined field value, and thus a proxy MUST NOT change the order of these field values when a message is forwarded. 

A `HashMap<String, String>` is currently used for response headers which will fail to handle this case correctly. That is,

```
Header-Name: Value1
Header-Name: Value2
```

will currently be collapsed into

```
Header-Name: Value 2
```

when sent from ProtocolServer, but it should either be passed with the duplicate fields intact, or collapsed into

```
Header-Name: Value1, Value2
```

before being sent across the IPC. if `Header-Name` is defined as a comma-separated list. I'm not sure what should happen for duplicate headers that are not comma-separated lists.

**Comment by awesomekling** [comment]
Indeed, for maximum fidelity, we should probably plumb the unaltered headers to the client side eventually. :)
